app/models/database/UserAnswers.scala

```diff
@@ -45,6 +46,26 @@ final case class UserAnswers(id: String, data: JsObject = Json.obj(), lastUpdate
     }
   }
 
+  def upsertFragment(businessId: BusinessId, dataFragment: JsObject): Try[UserAnswers] = {
+    val path = __ \ businessId.value
+    val upsertFragment: Reads[JsObject] = data
+      .validate((__ \ businessId.value).json.pick[JsObject])
+      .fold(
+        _ => path.json.put(dataFragment),
+        _ => path.json.update(__.read[JsObject].map(_ ++ dataFragment))
+      )
+
+    val updatedAnswers =
+      data.transform(upsertFragment) match {
+        case JsSuccess(jsValue, _) => Success(jsValue)
+        case JsError(errors)       => Failure(JsResultException(errors))
+      }
+
+    updatedAnswers.map { updatedData =>
+      copy(data = updatedData, lastUpdated = Instant.now)
+    }
+  }
```

Comment:
I may well me wrong here so do check, but is this code not equivalent to just doing the below:

```scala
    val existingAnswerData = (data \ businessId.value).asOpt[JsObject].getOrElse(JsObject.empty)
    val updatedData            = data + (businessId.value -> (existingAnswerData ++ dataFragment))

    copy(data = updatedData)
``` 

If so, it also gets rid of the Try.

Reply:
good one